toppra/dracula/topp_dracula_interface.py

```diff
@@ -190,14 +198,20 @@ class SplineInterpolator(Interpolator):
         # print("coeff shape", jnt_traj.cspl.c.shape)
         # print("break shape", jnt_traj.cspl.x.shape)
         # print("Done!")
-        return (xs_sample, ts_sample, sample_number)
+        return (xs_sample, ts_sample, sample_number, xds_sample, xdds_sample)
 
     else:
         #return knot points of the spline
         # print("cspl_x", jnt_traj.cspl.x.shape)
         # print("cspl_c", jnt_traj.cspl.c.shape)
         # print("Done!(spline)")
-        return (jnt_traj.cspl.x, jnt_traj.cspl.c)
+        jmax = 0.1
+        csplcp= copy.deepcopy(jnt_traj.cspl)
```

Comment:
Is it normal for functions in toppra to modify their arguments in-place? It would be more pythonic to return a modified version instead of mutating it.

Reply:
yep, that is what toppra does. it avoids making copies